corehq/apps/hqwebapp/forms.py

```diff
@@ -284,3 +285,39 @@ def get_context(self):
 
     def as_table(self):
         return render_to_string(self.template, self.get_context())
+
+
+class HQAuthenticationTokenForm(AuthenticationTokenForm):
+
+    def clean(self):
+        try:
+            cleaned_data = super(HQAuthenticationTokenForm, self).clean()
+        except ValidationError:
+            user_login_failed.send(sender=__name__, credentials={'username': self.user.username})
+            couch_user = CouchUser.get_by_username(self.user.username)
+            if couch_user and couch_user.is_locked_out() and couch_user.supports_lockout():
+                raise ValidationError(LOCKOUT_MESSAGE)
```

Comment:
this would tell user they have been locked out even when they don't submit the form successfully or the validation error itself is caused because user is locked out?